datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs

```diff
@@ -294,23 +294,9 @@ impl BloomFilterStatistics {
                 }
                 _ => true,
             },
-            // One more pattern matching since not all data types are supported
-            // inside of a Dictionary
-            ScalarValue::Dictionary(_, inner) => match inner.as_ref() {
-                ScalarValue::Int32(_)
-                | ScalarValue::Int64(_)
-                | ScalarValue::UInt32(_)
-                | ScalarValue::UInt64(_)
-                | ScalarValue::Float32(_)
-                | ScalarValue::Float64(_)
-                | ScalarValue::Utf8(_)
-                | ScalarValue::LargeUtf8(_)
-                | ScalarValue::Binary(_)
-                | ScalarValue::LargeBinary(_) => {
-                    BloomFilterStatistics::check_scalar(sbbf, inner, parquet_type)
-                }
-                _ => true,
-            },
+            ScalarValue::Dictionary(_, inner) => {
+                BloomFilterStatistics::check_scalar(sbbf, inner, parquet_type)
```

Comment:
The recursive function call will immediately do the same check, so there is no reason to also check the type before making the call